geomstats/visualization.py

```diff
@@ -490,6 +494,186 @@ def draw(self, ax, **kwargs):
         ax.scatter(translation[:, 0], translation[:, 1], s=16, **kwargs)
 
 
+class KendallSphere:
+    """Class used to plot points in Kendall shape space of 2D triangles."""
+
+    def __init__(self, points=None, point_type='pre-shape'):
+        self.points = []
+        self.point_type = point_type
+        self.ax = None
+        self.elev, self.azim = None, None
+
+        # equilateral triangle (north pole)
+        self.can = gs.array([[1., 0.],
+                             [-.5, gs.sqrt(3) / 2],
+                             [-.5, -gs.sqrt(3) / 2]]) / gs.sqrt(3)
+
+        # tangent direction toward isocele triangle at vertex A
+        self.ua = gs.array([[-1., 0.],
+                            [.5, gs.sqrt(3) / 2],
+                            [.5, -gs.sqrt(3) / 2]]) / gs.sqrt(3)
+
+        # tangent direction toward isocele triangle at vertex B
+        self.ub = gs.array([[.5, gs.sqrt(3) / 2],
+                            [.5, -gs.sqrt(3) / 2],
+                            [-1., 0.]]) / gs.sqrt(3)
+
+        # (ua,na) is a positively oriented othonormal basis of
+        # the horizontal space at north pole
+        self.na = self.ub - S32.ambient_metric.inner_product(
+            self.ub, self.ua) * self.ua
+        self.na = self.na / S32.ambient_metric.norm(self.na)
+
+        if points is not None:
+            self.add_points(points)
+
+    def set_ax(self, ax=None):
+        """Set axis."""
+        if ax is None:
+            ax = plt.subplot(111, projection='3d')
+
+        ax_s = .5
+        plt.setp(ax,
+                 xlim=(-ax_s, ax_s),
+                 ylim=(-ax_s, ax_s),
+                 zlim=(-ax_s, ax_s),
+                 xlabel='X', ylabel='Y', zlabel='Z')
+        self.ax = ax
+
+    def set_view(self, elev=30., azim=90.):
+        """Set azimuth and elevation angle."""
```

Comment:
Maybe add something like:
```
if self.ax is None:
        self.set_ax()
```
to avoid an error with `self.ax.view_init()` on Line 546?